app/models/solidus_paypal_braintree/gateway.rb

```diff
@@ -292,7 +295,7 @@ def transaction_options(source, options, submit_for_settlement = false)
       end
 
       params[:channel] = "Solidus"
-      params[:options] = { store_in_vault_on_success: true }
+      params[:options] = { store_in_vault_on_success: (preferred_paypal_flow == 'vault') }
```

Comment:
If we already make this change (what I think is great), we may want to extract that into a method, so stores don’t have to override this whole method if they don’t want to store the payment source, regardless of the pypal flow.

Reply:
@tvdeyen should we make this a preference on the gateway perhaps?

Reply:
The problem is that non devs are able to set preferences in the admin and store-in-vault only works if the flow is also vault.

Reply:
And ideally the customer should be able to choose. Neither the dev nor the admin, the user needs to be able to decide if she wants to store a token thats able to withdraw money without further verification.

Reply:
@tvdeyen agreed, it shouldn't be a preference. If it's configurable by the devs though should it be a method or some other kind of internal configuration option? For instance:
```ruby
SolidusPaypalBraintree.store_sources = true
```
I think this is better than the method, because theoretically methods are not a "clean" extension point - they have to be overridden with `prepend` or `class_eval`.

Re: letting users choose if they want to store the token, I think that's a good additional feature for later. How would you go about it? Some column on the order/payment model?

Reply:
@tvdeyen bump? :)

Reply:
> I think this is better than the method, because theoretically methods are not a "clean" extension point - they have to be overridden with `prepend` or `class_eval`.

@aldesantis a method is the only way I can imagine this will work with core Solidus right know as we need to know if the user wants to store a certain payment source.

Reply:
> Re: letting users choose if they want to store the token, I think that's a good additional feature for later. How would you go about it? Some column on the order/payment model?

@aldesantis yes a boolean on the payment source class toggled by a checkbox in the checkout view